tests/unit/test.program.js

```diff
@@ -242,13 +256,37 @@ describe('program.Program', () => {
       });
   });
 
+  it('checks for updates automatically', () => {
+    const root = path.join(__dirname, '..', '..');
+    const handler = spy();
+    const checkForAutomaticUpdates = sinon.stub();
+    const program = new Program(['run'])
+    .command('run', 'some command', handler);
+    return execProgram(program, {
+      checkForUpdates: checkForAutomaticUpdates,
+    })
+      .then(() => {
+        assert.equal(checkForAutomaticUpdates.firstCall.args[0].name,
+                      'web-ext');
+        let pkgFile = path.join(root, 'package.json');
+        return fs.readFile(pkgFile)
+          .then((pkgData) => {
```

Comment:
You should always chain promises instead of nest them. Otherwise you might as well use callbacks :) Chaining would look like:
````javascript
return execProgram()
  .then(() => {
    assert.equal(...);
    return fs.readFile(pkgFile);
  })
  .then((pkgFile) => {
    assert.equal(...);
  });
````